src/lib.rs

```diff
@@ -934,19 +930,102 @@ impl DstLayout {
     }
 }
 
-/// A trait which carries information about a type's layout that is used by the
-/// internals of this crate.
+/// Implements [`KnownLayout`].
+///
+/// This derive analyzes various qualities of a type's layout that are needed
+/// for some of zerocopy's APIs. It should be used whenver possible, and can be
+/// applied to structs, enums, and unions; e.g.:
+///
+/// ```
+/// # use zerocopy_derive::KnownLayout;
+/// #[derive(KnownLayout)]
+/// struct MyStruct {
+/// # /*
+///     ...
+/// # */
+/// }
+///
+/// #[derive(KnownLayout)]
+/// enum MyEnum {
+/// #   V00,
+/// # /*
+///     ...
+/// # */
+/// }
+///
+/// #[derive(KnownLayout)]
+/// union MyUnion {
+/// #   variant: u8,
+/// # /*
+///     ...
+/// # */
+/// }
+/// ```
+///
+/// # Limitations
+///
+/// This derive cannot currently be applied to unsized structs with the
+/// default-`repr`.
```

Comment:
Not sure that "default-`repr`" is a standard term. Maybe "without an explicit `repr` attribute" or something?

Reply:
Done